lib/types/src/module_hash.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt::{self, Display, Formatter};
+
+use rand::RngCore;
+use rkyv::{Archive, CheckBytes, Deserialize as RkyvDeserialize, Serialize as RkyvSerialize};
+#[cfg(feature = "enable-serde")]
+use serde::{Deserialize, Serialize};
+use sha2::Digest;
+
+/// Hashing algorithm to be used for the module info
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum HashAlgorithm {
+    /// Sha256
+    Sha256,
+    /// XXHash
+    XXHash,
+}
+
+/// The hash of a WebAssembly module.
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    RkyvSerialize,
+    RkyvDeserialize,
+    Archive,
+)]
+#[archive_attr(derive(CheckBytes, Debug))]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub enum ModuleHash {
+    /// xxhash
+    XXHash([u8; 8]),
+
+    /// sha256
+    Sha256([u8; 32]),
+}
+
+impl ModuleHash {
+    /// Create a new [`ModuleHash`] from the raw xxhash hash.
+    pub fn xxhash_from_bytes(key: [u8; 8]) -> Self {
+        Self::XXHash(key)
+    }
+
+    /// Create a new [`ModuleHash`] from the raw sha256 hash.
+    pub fn sha256_from_bytes(key: [u8; 32]) -> Self {
+        Self::Sha256(key)
+    }
+
+    /// Creates a random xxhash for the module
+    pub fn xxhash_random() -> Self {
+        let mut rand = rand::thread_rng();
+        let mut key = [0u8; 8];
+        rand.fill_bytes(&mut key);
+        Self::xxhash_from_bytes(key)
+    }
+
+    /// Creates a random sha256 hash for the module
+    pub fn sha256_random() -> Self {
+        let mut rand = rand::thread_rng();
+        let mut key = [0u8; 32];
+        rand.fill_bytes(&mut key);
+        Self::sha256_from_bytes(key)
+    }
```

Comment:
Can we remove this methods? They are not used anywhere, and it will help us to remove the dependency on rand

Reply:
Correction: `ModuleHash::xxhash_random` is used only in wasix.
Lets move it just to wasix.

It will help slim down dependencies for wasmer-types and wasmer-js